src/mono/mono/component/hot_reload.c

```diff
@@ -1217,8 +1221,8 @@ hot_reload_get_method_debug_information (MonoImage *image_dppdb, int idx)
 			int token_index = mono_metadata_token_index (map_token);
 			if (token_index == idx)	{
 				MonoDebugInformationEnc *encDebugInfo = g_new0 (MonoDebugInformationEnc, 1);
-				encDebugInfo->idx = i;
-				encDebugInfo->image = image_dppdb;				
+				encDebugInfo->idx = i + 1;
+				encDebugInfo->ppdb_file = mono_create_ppdb_file (image_dppdb, FALSE);
```

Comment:
This is going to create a new MonoPPDBFile for each row in the methodbody table, isn't it? 

Maybe it's better to create the MonoPPDBFile when we first apply the delta (maybe store it in the DeltaInfo), and then just reference it from the MonoDebugInformatioEnc structs when you do the lookup here.